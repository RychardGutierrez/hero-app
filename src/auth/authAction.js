export const actionContex = (type, payload) => ({
  type,
  payload,
});
