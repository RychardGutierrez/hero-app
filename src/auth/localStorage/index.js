export const setStorage = (nameStorage, data) =>
  localStorage.setItem(nameStorage, JSON.stringify(data));

export const getStorage = (nameStorage) =>
  JSON.parse(localStorage.getItem(nameStorage));
