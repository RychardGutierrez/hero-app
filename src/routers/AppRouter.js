import { Routes, Route, BrowserRouter } from 'react-router-dom';

import LoginScreen from './../login/container/LoginScreen';
import DashboardRoutes from './DashboardRoutes';
import DoNotFound from './../UI/components/DoNotFound';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';

const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route
          path="/login"
          element={
            <PublicRoute>
              <LoginScreen />
            </PublicRoute>
          }
        ></Route>

        {/* <Route path="/login" element={<LoginScreen />} /> */}
        <Route path="/404" element={<DoNotFound />} />
        <Route
          path="/*"
          element={
            <PrivateRoute>
              <DashboardRoutes />
            </PrivateRoute>
          }
        ></Route>

        {/* <Route path="/*" element={<DashboardRoutes />} /> */}
      </Routes>
    </BrowserRouter>
  );
};

export default AppRouter;
