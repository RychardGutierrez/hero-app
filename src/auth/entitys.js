import { getStorage } from './localStorage';
import { STORAGE_USER } from './localStorage/constants';

export const initReducer = () => {
  const initState = getStorage(STORAGE_USER) || {
    logged: false,
  };

  return initState;
};
