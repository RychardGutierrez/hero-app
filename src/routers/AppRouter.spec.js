import { mount } from 'enzyme';
import { AuthContext } from '../auth/authContext';
import AppRouter from './AppRouter';

describe('AppRouter', () => {
  it('Should return login if the user is do not authenticate', () => {
    const conextValue = {
      user: {
        logged: false,
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={conextValue}>
        <AppRouter />
      </AuthContext.Provider>,
    );

    expect(wrapper.find('h1').text().trim()).toBe('Login');
    expect(wrapper).toMatchSnapshot();
  });

  it('Should return DasboardRoutes when the user is authenticate', () => {
    const conextValue = {
      user: {
        logged: true,
        name: 'Lucas',
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={conextValue}>
        <AppRouter />
      </AuthContext.Provider>,
    );

    expect(wrapper.find('.navbar').exists()).toBeTruthy();

    expect(wrapper.find('.nav-item.nav-link.btn').text().trim()).toBe('Logout');
    expect(wrapper.find('.nav-item.nav-link.text-info').text().trim()).toBe(
      'LUCAS',
    );

    expect(wrapper.find('h1').text().trim()).toBe('Welcome to Heroes App');

    expect(wrapper).toMatchSnapshot();
  });
});
