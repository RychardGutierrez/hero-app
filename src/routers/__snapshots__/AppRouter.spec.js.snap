// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter Should return DasboardRoutes when the user is authenticate 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark  bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Associations
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link false"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link false"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link false"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          LUCAS
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Welcome to Heroes App
      </h1>
    </div>
  </div>,
]
`;

exports[`AppRouter Should return login if the user is do not authenticate 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
